include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+/**
+ * Implements Google Calendar Syncing
+ *
+ * @license https://raw.githubusercontent.com/salesagility/SuiteCRM/master/LICENSE.txt GNU Affero General Public License version 3
+ * @author Benjamin Long <ben@offsite.guru>
+ */
+
+class GoogleSync
+{
+
+    /** @var array An array of user id's we are going to sync for */
+    public $users = array();
+
+    /** @var object The User we're currently working with */
+    public $workingUser;
+
+    /** @var object The Google client object for the current sync job */
+    public $gClient;
+
+    /** @var object The Google Calendar Service Object */
+    public $gService;
+
+    /** @var array The Google AuthcConfig json */
+    public $authJson = array();
+
+    /** @var string The local timezone */
+    public $timezone;
+
+    /** @var string The Calendar ID */
+    public $calendarId;
+
+    /** @var array An array of SuiteCRM meeting id's that we've already synced this session */
+    public $syncedList = array();
+
+    /** @var object A Database Instance */
+    private $db;
+
+    /**
+     * Class Constructor
+     */
+    public function __construct()
+    {
+        if (isset($_SERVER['GSYNC_LOGLEVEL']))
+        {
+            $GLOBALS['log']->setLevel($_SERVER['GSYNC_LOGLEVEL']);
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Log Level Set to: ' . $_SERVER['GSYNC_LOGLEVEL']);
+        }
+        $this->timezone = date_default_timezone_get(); // This defaults to the server timezone. Overridden later.
+        $this->authJson = $this->getAuthJson();
+        $this->db = DBManagerFactory::getInstance();
+        $GLOBALS['log']->debug(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . '__construct');
+    }
+
+    /**
+     * Gets the auth json string from the system
+     *
+     * @return string|false json on success, false on failure
+     */
+    public function getAuthJson()
+    {
+        global $sugar_config;
+        if (!$authJson_local = json_decode(base64_decode($sugar_config['google_auth_json']), true))
+        {
+            // The authconfig json string is invalid json
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Invalid AuthConfig JSON');
+            return false;
+        }
+        else {
```

Comment:
Minor Issue: PSR 2 - else must be on the same line as the closing brace '}'. Consider using a psr 2 auto formating tool.

Reply:
https://docs.suitecrm.com/community/contributing-code/coding-standards/ should be updated to reflect this. It shows else on a new line.
I'll fix this and follow PSR-2.

Reply:
Thank You, We will update the documentation to follow this pattern.